subsys/net/lib/sockets/sockets.c

```diff
@@ -915,6 +918,28 @@ static int zsock_poll_prepare_ctx(struct net_context *ctx,
 		(*pev)++;
 	}
 
+	if (pfd->events & ZSOCK_POLLOUT) {
+		if (*pev == pev_end) {
+			errno = ENOMEM;
+			return -1;
+		}
+
+#if defined(CONFIG_NET_TCP)
+		if (net_context_get_ip_proto(ctx) == IPPROTO_TCP &&
+		    ctx->tcp && net_tcp_get_recv_wnd(ctx->tcp) == 0) {
+			;
+		} else
+#endif
+		{
+			errno = EINPROGRESS;
```

Comment:
What's the meaning of this? You cannot return arbitrary errors from prepare_ctx() vmethod, such an error will be fatal for poll(), and fatal errors for poll() should be minimized (in particular, problem with a *single* socket should never lead to abort of entire poll())

Reply:
We can certainly return EALREADY here too, the patch just returns EINPROGRESS just to differentiate POLLIN and POLLOUT.

Reply:
> We can certainly return EALREADY here too, the patch just returns EINPROGRESS just to differentiate POLLIN and POLLOUT.

There's no need to introduce additional notions. EALREADY is the API contract for signaling behavior of "Ignore the incoming timeout param, and perform underlying k_poll() call with K_NO_WAIT, but be sure to still perform". If this fact is not clear enough and you see where it can be documented better (perhaps, in a code comments, as this is internal API contract), then +1 for doing that.